Added stagger and grid_mapping attributes to diagnostics

lpilz opened the pull request:
<!-- Thanks for submitting a PR, your contribution is really appreciated! -->
<!-- Unless your change is trivial, please create an issue to discuss the change before creating a PR -->
<!-- Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->

## Change Summary

I found some attributes to be missing from the diagnostics. Shouldn't they be included?

## Related issue number

<!-- Are there any issues opened that will be resolved by merging this change? -->
<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

## Checklist


- [x] Tests pass on CI


<!--
Please add any other relevant info below:
-->

kmpaul commented:
@lpilz:  Thanks for this.  Is there anything else that is missing?  Or should we merge this and "catch more problems later"?

jthielen reviewed (changes requested):
Good catch, and implementation looks good to me! The addition of `grid_mapping` is definitely needed. `stagger` is something WRF-specific that is redundant with the coordinate information (https://github.com/ncar-xdev/xwrf/issues/15#issue-1059026669), so I don't think it's really needed to be added here, but it doesn't hurt either given that we aren't stripping it from the existing fields.

That being said, it would be good to have a basic test added prior to merging this, especially given that this slipped by previously.

And, yes, I concur with addressing just the known problems and merging now rather than trying to speculate and exhaust all possible issues here!

lpilz commented:
I removed the manual addition of the `grid_mapping` attr from the diagnostics calculation and rather moved the assignment of the projection coordinate to the end of the `postprocess()` pipleine. Also added some tests. Ready for a re-review :)

kmpaul commented on tests/test_postprocess.py:
```diff
@@ -60,6 +60,17 @@ def test_include_projection_coordinates(sample_dataset):
+            assert dataset[var].attrs['grid_mapping'] == 'wrf_projection'
+
+
 @importorskip('xgcm')
```
```suggestion
import sys
@importorskip('xgcm')
@pytest.mark.skipif(sys.version_info < (3, 9), reason="xgcm only works with Python 3.9+")
```

kmpaul reviewed (changes requested):
This suggestion will fail linting, but pre-commit.ci should fix it.

kmpaul commented on tests/test_postprocess.py:
```diff
@@ -60,7 +60,22 @@ def test_include_projection_coordinates(sample_dataset):
+
+import sys
+
+
 @importorskip('xgcm')
+@pytest.mark.skipif(sys.version_info < (3, 9), reason='xgcm only works with Python 3.9+')
```
```suggestion
@pytest.mark.skipif(sys.version_info < (3, 9), reason='xgcm only works with Python 3.9+')
@importorskip('xgcm')
```

kmpaul reviewed (commented):
Didn't work as planned.  Try reversing order.

kmpaul commented:
Ok.  I'm sorry for mucking around, but it looks like the `importorskip` function is going to need to be modified to accommodate this use case.  I'll try to mock something up and submit it to this PR.

kmpaul reviewed (approved):
Okay.  I think this fixes the issue and still maintains the basic purpose of the `importorskip` function.
